src/core/thread/indirect_sender.cpp

```diff
@@ -163,14 +163,15 @@ void IndirectSender::ClearAllMessagesForSleepyChild(Child &aChild)
     mSourceMatchController.ResetMessageCount(aChild);
 
     mDataPollHandler.RequestFrameChange(DataPollHandler::kPurgeFrame, aChild);
-#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
     mCslTxScheduler.Update();
 #endif
 
 exit:
     return;
 }
 
+#if OPENTHREAD_FTD
 void IndirectSender::SetChildUseShortAddress(Child &aChild, bool aUseShortAddress)
 {
```

Comment:
```suggestion
void IndirectSender::SetChildUseShortAddress(Child &aChild, bool aUseShortAddress)
{
#if OPENTHREAD_FTD
```

How about making `SetChildUseShortAddress` an empty function when `OPENTHREAD_FTD` is not defined and does not use macro where `SetChildUseShortAddress` is called?

Reply:
Well, I would prefer to not define it at all in MTD case for: 1. smaller code size 2. more efficient And I think this helps us to differentiate Child for normal case and for s2s case.